src/Microsoft.Diagnostics.Monitoring.RestServer/RequestThrottling/Throttling.cs

```diff
@@ -0,0 +1,82 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Logging;
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Diagnostics.Monitoring.RestServer
+{
+    /// <summary>
+    /// Limits the amount of requests that can be sent to the server.
+    /// - The current rate limits are based on concurrent requests to the server from any source on a per endpoint basis.
+    /// - Note we do not use Microsoft.AspNetCore.ConcurrencyLimiter because it works over the whole application instead of per endpoint.
+    /// - In the future, we may want to switch to https://github.com/dotnet/aspnetcore/issues/29933
+    /// TODO For asp.net 2.1, this would be implemented as an ActionFilter. For 3.1+, we use an endpoints + middleware
+    /// </summary>
+    internal sealed class Throttling
```

Comment:
I think this should be moved to dotnet-monitor (instead of the RESTServer assembly) since how the throttling is added to the pipeline is application specific. Other applications might have their on throttle mechanism.